packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -788,11 +788,18 @@ class Tree {
       if (
         current !== node &&
         node.nodeToRestore &&
-        current.nodeToRestore &&
-        node.scopeRef.current &&
-        isElementInScope(current.nodeToRestore, node.scopeRef.current)
+        (
+          // current.nodeToRestore might be undefined when the current node has just been added.
+          !current.nodeToRestore ||
+          (
+            current.nodeToRestore &&
```

Comment:
Do you have to check for `current.nodeToRestore`? Isn't it implied on this side of the `||`?